### PR TITLE
Rename HorizontalStack to InlineStack

### LINKED ISSUE #9983: HorizontalStack to InlineStack
- [x] Rename components throughout the codebase
- [x] Improve documentation
- [x] Add migrations to migration guide

### PR DESCRIPTION
**Pull request by kyledurand**
### WHY are these changes introduced?

The first part of https://github.com/Shopify/polaris/issues/9983

Will follow up with docs improvement

### How to 🎩

🖥 [Local development instructions](https://github.com/Shopify/polaris/blob/main/README.md#local-development)
🗒 [General tophatting guidelines](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting.md)
📄 [Changelog guidelines](https://github.com/Shopify/polaris/blob/main/.github/CONTRIBUTING.md#changelog)

<details>
<summary>Copy-paste this code in <code>playground/Playground.tsx</code>:</summary>

```jsx
import React from 'react';
import {Page} from '../src';

export function Playground() {
  return (
    <Page title="Playground">
      {/* Add the code you want to test in here */}
    </Page>
  );
}
```

</details>

### 🎩 checklist

- [ ] Tested on [mobile](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting.md#cross-browser-testing)
- [ ] Tested on [multiple browsers](https://help.shopify.com/en/manual/shopify-admin/supported-browsers)
- [ ] Tested for [accessibility](https://github.com/Shopify/polaris/blob/main/documentation/Accessibility%20testing.md)
- [ ] Updated the component's `README.md` with documentation changes
- [ ] [Tophatted documentation](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting%20documentation.md) changes in the style guide

### CODE REVIEW

File: polaris.shopify.com/next.config.js [hunk 1]
```diff
@@ -397,11 +397,12 @@ const layoutAndStructure = [
...
-    source: '/components/inline',
-    destination: '/components/layout-and-structure/horizontal-stack',
-    permanent: false,
-  },
+  // TODO: Re-enable this redirect once the new inline stack component is released
+  // {
+  //   source: '/components/layout-and-structure/horizontal-stack',
+  //   destination: '/components/layout-and-structure/inline-stack',
+  //   permanent: false,
+  // },
```
Comment (yurm04) [inline polaris.shopify.com/next.config.js:405]: could you not do this now if the doc updates will only be published once we merge in `next` and release v12?

**Review by yurm04 — APPROVED** [review]
Looks good 👍 just left one question that isn't blocking

File: polaris.shopify.com/next.config.js (hunk 1 above)
Comment (kyledurand) [reply polaris.shopify.com/next.config.js:405]: Not sure why but this was causing build errors. The sitemap failed thinking this was still a valid route even though it's only supposed to be a redirect. I'll dig in further when this is closer to shipping. I don't want to disrupt other se23 workflows

Comment (kyledurand) [reply polaris.shopify.com/next.config.js:405]: Going to leave this commented out and follow up https://github.com/Shopify/polaris/issues/10107

**Pull request merged**